### DISCUSSION TITLE
Merging chapterized files

### DISCUSSION THREAD

**Discussion by AudiobookListener** [yermak/AudioBookConverter#349] [Q&A] [answered]
When merging M4B files, is it expected that they will loose existing chapters, and the file contents become the new chapter marks?

Many early M4B audiobooks were split into multiple chaptered M4Bs due to file-size limits of older storage and playback devices.

I've tried working around this by splitting them by chapter then merging, but can't seem to get chapter-split working on M4Bs, not even when I make a cue file using another utility.

**Comment by yermak** [comment]
Hi,
Let me try to understand what exactly you are trying to achieve...

You have set of m4b files with chapters, you are trying to merge them all together into single m4b and expect to have single m4b with new chapters where each chapter will be based on m4b file, but not on chapters inside them.

If this assumption is correct. Then I suggest following workaround, as default behaviour of the program to preserve chapters from original m4b files, it could be time consuming...

After you import directory with you m4b files you see list of files, you press chapters and you see a lot of chapters imported from your m4b and it seems you lost file information... and all chapters are renumbered.

However, what you could do:
- Watch names of original chapters in details column, depending how they were called you may easy identify that next book is started.
- Select first/any chapter, right click -> Edit -> Delete information in text field and select two fields: Chapter No and file_name from dropdown list. IMPORTANT: select apply to all chapters. Now all you chapters are renamed according to new rule: chapter Number and file name.

Whichever of 2 options you used (i.e. in my test example i just visually saw that new file started, but i tested second option as well). you now need to select range of chapters from the SAME file with "Shift" -> Right Click -> Combine -> you will get SINGLE chapter per file. Repeat this for each original m4b file.


Please let me know if this helps.

**Comment by yermak** [comment, marked as answer]
Some additional thoughts...

If you importing multiple m4b and would like their chapter information to be lost and source m4b just become chapter. You need to combine content of all chapters for each file into single chapter, then use split by parts, which gives you single book where chapters represent source files.
